admin/tabs/roles/details/members/RoleMembersModel.ts

```diff
@@ -0,0 +1,205 @@
+            ROLE: roles.length
+        };
+    }
+
+    @computed
+    get effectiveCounts(): Record<RoleMemberType, number> {
+        if (!this.selectedRole) return null;
+        const {effectiveUsers, effectiveDirectoryGroups, effectiveRoles} = this.selectedRole;
+        return {
+            USER: effectiveUsers.length,
+            DIRECTORY_GROUP: effectiveDirectoryGroups.length,
+            ROLE: effectiveRoles.length
+        };
+    }
+
+    get activeTabId(): string {
+        return this.roleDetailsModel.tabContainerModel.activeTabId;
+    }
+
+    setActiveTabId(id: string) {
+        this.roleDetailsModel.tabContainerModel.activateTab(id);
+    }
+
+    override onLinked() {
+        this.gridModel = this.createGridModel();
+        this.addReaction({
+            track: () => this.selectedRole,
+            run: role => this.loadGridData(role),
+            fireImmediately: true,
+            debounce: 100
+        });
+    }
+
+    // -------------------------------
+    // Implementation
+    // -------------------------------
+
+    private loadGridData(role: HoistRole) {
+        if (!role) {
+            this.gridModel.clear();
+            return;
+        }
+
+        if (this.props.showEffective) {
+            this.gridModel.loadData([
+                // 1 - Users
+                ...role.effectiveUsers.map(it => ({
+                    name: it.name,
+                    sources: this.sortThisRoleFirst(it.sources),
+                    type: RoleMembersModel.types.USER
+                })),
+
+                // 2 - Directory Groups
+                ...role.effectiveDirectoryGroups.map(it => ({
+                    name: it.name,
+                    sources: this.sortThisRoleFirst(it.sourceRoles.map(role => ({role}))),
+                    type: RoleMembersModel.types.DIRECTORY_GROUP
+                })),
+
+                // 3 - Roles
+                ...role.effectiveRoles.map(it => ({
+                    name: it.name,
+                    sources: this.sortThisRoleFirst(it.sourceRoles.map(role => ({role}))),
+                    type: RoleMembersModel.types.ROLE
+                }))
+            ]);
+        } else {
+            this.gridModel.loadData(
+                role.members.map(it => ({...it, type: RoleMembersModel.types[it.type]}))
+            );
+        }
+    }
+
+    private createGridModel(): GridModel {
+        const {types} = RoleMembersModel,
+            {showEffective} = this.props;
+        return new GridModel({
+            store: {
+                fields: [
+                    {name: 'name', type: 'string'},
+                    {name: 'type', type: 'string'},
+                    {name: 'sources', displayName: 'Assigned Via', type: 'json'}
+                ],
+                idSpec: data => `${this.selectedRole.name}:${data.type}:${data.name}`
+            },
+            emptyText: 'This role has no members.',
+            filterModel: true,
```

Comment:
Thinking this is overkill to have grid filters on this grid, plus there is some issue with the sources JSON field - would remove, unless you think they would be especially useful and we can fix that issue

Reply:
Ah yes agreed, will remove